### DISCUSSION TITLE
How can i add a new language

### DISCUSSION THREAD

**Discussion by wassim31** [dzcode-io/kuliya#37] [Ideas]
It will C language where i will add my implementation of some searching & sorting algorithms using C ( they're well explained in readme :smile: )

**Comment by ZibanPirate** [comment]
All you need to do is create a `cpp` folder, inside that folder you're the king, and you can access the data in `../_data` folder

**Comment by ZibanPirate** [comment]
essentially, you need two functionalities:
1. convert the data inside `_data` folder, from folders/files into one single object preferably JSON, but if there's a better format for C++ then you're the expert, see [JS implementation](https://github.com/dzcode-io/kuliya/blob/main/javascript/src/_config/prepare.ts) and also [Java implementation](https://github.com/dzcode-io/kuliya/blob/main/java/src/main/java/io/dzcode/kuliya/PrepareData.java) by @nacersalaheddine 
2. expose the first method of the API, `getNodeByPath` or `GetNodeByPath` to respect C++'s case style, see [JS implementation](https://github.com/dzcode-io/kuliya/blob/main/javascript/src/getNodeByPath/index.ts) and [Java implementation](https://github.com/dzcode-io/kuliya/blob/main/java/src/main/java/io/dzcode/kuliya/Search.java) again by the awesome @nacersalaheddine 

note that the first functionality (preparing the data) should happen on the build/compile/release time, which generate a formatted file inside `cpp` folder, your code should consume this file instead of the `_data` folder

**Reply by omdxp** [reply]
> ... `GetNodeByPath` to respect C++'s case style, ...

Correction, to respect .NET case style or Windows enthusiasts, you can write with snake case if you want.
